app/routes.py

```diff
@@ -0,0 +1,381 @@
+from app import db
+from flask import Blueprint, jsonify, request
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+from datetime import datetime, timedelta
+import os
+
+
+videos_bp = Blueprint("videos_bp", __name__, url_prefix="/videos")
+customers_bp = Blueprint("customers_bp", __name__, url_prefix="/customers")
+rentals_bp = Blueprint("rentals_bp", __name__, url_prefix="/rentals")
+
+# VIDEO ROUTES
+@videos_bp.route("", methods=["POST", "GET"])
+def handle_videos():
+    # POST REQUEST
+    if request.method == "POST":
+        video_request_body = request.get_json()
+        
+        if "title" not in video_request_body:
+            return jsonify({"details": "Request body must include title."}), 400
+        elif "release_date" not in video_request_body:
+            return jsonify({"details": "Request body must include release_date."}), 400
+        elif "total_inventory" not in video_request_body:
+            return jsonify({"details": "Request body must include total_inventory."}), 400
+    
+        new_video = Video(
+            title=video_request_body["title"],
+            release_date=video_request_body["release_date"],
+            total_inventory=video_request_body["total_inventory"]
+        )
+
+        db.session.add(new_video)
+        db.session.commit()
+
+        new_video_response = new_video.get_video_dict()
+
+        return jsonify(new_video_response), 201   
+    # GET REQUEST
+    elif request.method == "GET":
+        video_title_query = request.args.get("title")
+        if video_title_query:
+            videos = Video.query.filter(Video.title.contains(video_title_query))
+        else:
+            videos = Video.query.all()
+
+        #OPTIONAL ENHANCEMENT (Query Params)
+        # ---------------------------------
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            videos = Video.query.order_by(Video.id.asc())
+            videos = Video.query.order_by(Video.title.asc())
+        else:
+            videos = Video.query.all()
+
+        video_response = [video.get_video_dict() for video in videos]
+        
+        if video_response == []:
+            return jsonify(video_response), 200
+
+        return jsonify(video_response), 200
+
+# GET, PUT, DELETE ONE VIDEO AT A TIME
+@videos_bp.route("/<video_id>", methods=["GET", "PUT", "DELETE"])
+def handle_one_video_at_a_time(video_id):
+    if not video_id.isnumeric():
+        return jsonify(None), 400
```

Comment:
another suggestion is to do a try/except to handle if the video id is not an integer 
```suggestion
   try:
        video_id = int(video_id)
    except ValueError:
        return f"Invalid video id '{video_id}'",400
```